### DISCUSSION TITLE
Can the sionna support cell-free? (multiple base station serving a single user with the same frequency)

### DISCUSSION THREAD

**Discussion by JackokieZhao** [NVlabs/sionna#269] [Q&A]
I want to simulate cell-free MIMO, in which multiple base station can serve a single user with the same frequency. But I found the NR simulation cannot support multiple base station. I really don't know how to modify the sionna. In my opinion, it might be a general demand for lots of cell-free researcher.

**Comment by jhoydis** [comment]
Hi,

I believe that you would simply need to change the channel model to implement cell-free MIMO. That means that you would simulate the channels between multiple base stations and users and then combine the channel impulse responses so that they represent a large distributed antenna array.

When you check, for example, the [API documentation of the 38.901 UMI channel model](https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.tr38901.UMi), you can see that the shape of the channel coefficients `a` is `[batch size, num_rx, num_rx_ant, num_tx, num_tx_ant, num_paths, num_time_steps]`. If you define multiple base stations, i.e., num_rx>1, you could simply reshape `a` to 
`[batch size, 1, num_rx*num_rx_ant, num_tx, num_tx_ant, num_paths, num_time_steps]` to combine them to a large distributed receiver. Also the delays `tau` would need to be reshaped from `[batch size, num_rx, num_tx, num_paths]` to `[batch size, 1, num_rx*num_tx, num_paths]`.

This approach is independent of the channel model (stochastic as well as ray tracing) and should work with the `NR` module out-of-the-box. You might want to be careful about how you normalise the channel impulse responses though (e.g., do you want to have path loss and shadow fading or not). 

Hope that helps.

**Reply by JackokieZhao** [reply]
Hi, thanks for your greate help! In addition, the base station might need more work, such as joint precoding, MCS. I don't know whether the StreamManagement support multiple base station. In addition, whether the PUSCH is equal to PDSCH? 

Thanks a lot!

**Reply by jhoydis** [reply]
If you want to work on the downlink, you cannot use PUSCH. PDSCH is not implemented in Sionna, yet. Do you really need to do 5G-compliant simulations for your research work? You could otherwise build a much simpler [OFDM system](https://nvlabs.github.io/sionna/api/ofdm.html). ``StreamManagement`` supports multiple base stations but this would only be needed if you want to have multiple interfering distributed base stations.

**Reply by JackokieZhao** [reply]
I want to evaluate the precoding and MCS performance for multiple base station VS single user. If the OFDM module can work, I would be try it. Thanks!

**Reply by NowhereMan-042** [reply]
Hi there,
Just a related query. I'm interested in simulating cell-free massive MIMO for my research. I was wondering if Sionna Ray tracing currently supports simulating this type of scenario with multiple base stations?

**Reply by SebastianCa** [reply]
Hi @NowhereMan-042,
yes, you can run the ray tracer for multiple base-stations. See [Paths.cir() function](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Paths.cir) for further details and the [Ray Tracing Tutorial Notebook](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html#Site-specifc-Link-Level-Simulations) shows an example how to place multiple receivers.
